### PR TITLE
 Add Update task query inside rtk query api

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
### Description of Change
-  Add Update task query inside rtk query api
### Associated Ticket Link:
- https://trello.com/c/uupY6ZVo

### CODE REVIEW

File: shared/redux/rtk-apis/tasksAPI.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
 
 import { TTask } from "@/shared/typedefs/types";
+import { TCreateTaskPayload, TUpdateTaskPayload } from "./tasksAPI.types";
```
Comment (ishmam-iqbal-sazim) [inline shared/redux/rtk-apis/tasksAPI.ts:4]: Line gap between @ and relative imports

File: shared/redux/rtk-apis/tasksAPI.types.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { EPriority, ETaskStatus } from "@/shared/typedefs/enums";
+
+export type TCreateTaskPayload = {
+  title: string;
+  description?: string;
+  dueDate?: Date | null;
+  priority: EPriority;
+  status: ETaskStatus;
+};
```
Comment (ishmam-iqbal-sazim) [inline shared/redux/rtk-apis/tasksAPI.types.ts:9]: Should not be seeing changes from other PRs. Please be mindful of rebasing in the future

File: shared/redux/rtk-apis/tasksAPI.types.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+export type TUpdateTaskPayload = {
```
Comment (ishmam-iqbal-sazim) [inline shared/redux/rtk-apis/tasksAPI.types.ts:11]:
```suggestion
export type TUpdateTaskDto = {
```

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**